### PR TITLE
encode the password and test RedisURI for encode password

### LINKED ISSUE #235: wrong password with old api in 3.9
In the old api, it will convert `io.vertx.redis.RedisOptions` to a connectionString like:

```
redis://localhost:6379/0?password=xxx
```

but if my password contains '#', it will get a wrong password in `RedisURI`, it seem like that the password dos not encode. e.g

```
redis://localhost:6379/0?password=abc#123
```
The real password is `abc#123`，but the param in the query is `abc`

### PR DESCRIPTION
**Pull request by okou19900722**
fixed #235

### CODE REVIEW

File: src/test/java/io/vertx/redis/client/impl/RedisURITest.java [hunk 1]
```diff
@@ -33,6 +36,14 @@ public void testTwoPasswordsAreGiven() {
     Assert.assertEquals("Password is not correct", "pass", redisURI.password());
   }
 
+  @Test
+  public void testPasswordWithEncodeInQuery() throws UnsupportedEncodingException {
```
Comment (gaol) [inline src/test/java/io/vertx/redis/client/impl/RedisURITest.java:40]: This test passes even without the fix. Actually I believe this is not a bug.

Comment (okou19900722) [reply src/test/java/io/vertx/redis/client/impl/RedisURITest.java:40]: have you run the test? a new test was commit.you can review it

Comment (gaol) [reply src/test/java/io/vertx/redis/client/impl/RedisURITest.java:40]: After contact offline, I understand the problem now. Yes it should be a bug in the old API :)

**Comment by pmlopes** [comment]
Thanks!

**Pull request merged**